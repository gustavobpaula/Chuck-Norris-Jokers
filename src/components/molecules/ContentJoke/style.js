import styled from 'styled-components';
import designSystem from '../../../designSystem';

export const Joke = styled.article`
  background-color: ${designSystem.get('colors.color2')};
  border-radius: 20px;
  margin-top: 50px;
  padding: 50px;
  text-align: center;
  width: 100%;
`;
