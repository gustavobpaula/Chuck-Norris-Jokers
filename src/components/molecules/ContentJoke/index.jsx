import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '../../atoms';
import * as Styled from './style';

const ContentJoke = ({ children }) => (
  <Styled.Joke>
    <Typography variant="joke" component="h1">
      {children}
    </Typography>
  </Styled.Joke>
);

ContentJoke.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ContentJoke;
