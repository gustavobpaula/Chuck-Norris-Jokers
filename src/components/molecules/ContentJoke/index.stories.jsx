import React from 'react';
import ContentJoke from './index';

export default {
  component: ContentJoke,
  title: 'Molecules/ContentJoke',
};

export const element = () => (
  <ContentJoke>
    A Handicapped parking sign does not signify that this spot is for
    handicapped people. It is actually in fact a warning, that the spot belongs
    to Chuck Norris and that you will be handicapped if you park there
  </ContentJoke>
);

element.story = {
  name: 'Default',
};
