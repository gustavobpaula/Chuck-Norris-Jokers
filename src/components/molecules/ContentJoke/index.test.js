import React from 'react';
import { mount } from 'enzyme';
import ContentJoke from './index';

describe('<ContentJoke />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(
      <ContentJoke>
        A Handicapped parking sign does not signify that this spot is for
        handicapped people. It is actually in fact a warning, that the spot
        belongs to Chuck Norris and that you will be handicapped if you park
        there
      </ContentJoke>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
