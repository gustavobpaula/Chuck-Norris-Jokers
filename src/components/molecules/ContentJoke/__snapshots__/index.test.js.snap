// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentJoke /> should mount properly 1`] = `
<ContentJoke>
  <style__Joke>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "style__Joke-sc-1tgzdwc-0",
            "isStatic": false,
            "lastClassName": "OPlMV",
            "rules": Array [
              "background-color:",
              "#5588a3",
              ";border-radius:20px;margin-top:50px;padding:50px;text-align:center;width:100%;",
            ],
          },
          "displayName": "style__Joke",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "style__Joke-sc-1tgzdwc-0",
          "target": "article",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <article
        className="style__Joke-sc-1tgzdwc-0 OPlMV"
      >
        <Typography
          component="h1"
          variant="joke"
        >
          <style__Typography
            as="h1"
            variant="joke"
          >
            <StyledComponent
              as="h1"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "style__Typography-sc-8vyb1l-0",
                    "isStatic": false,
                    "lastClassName": "eeZxzb",
                    "rules": Array [
                      [Function],
                    ],
                  },
                  "displayName": "style__Typography",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "style__Typography-sc-8vyb1l-0",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              variant="joke"
            >
              <h1
                className="style__Typography-sc-8vyb1l-0 eeZxzb"
              >
                A Handicapped parking sign does not signify that this spot is for handicapped people. It is actually in fact a warning, that the spot belongs to Chuck Norris and that you will be handicapped if you park there
              </h1>
            </StyledComponent>
          </style__Typography>
        </Typography>
      </article>
    </StyledComponent>
  </style__Joke>
</ContentJoke>
`;
